llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3976,6 +4059,10 @@ Instruction *InstCombinerImpl::visitOr(BinaryOperator &I) {
     if (Value *Res = foldBitmaskMul(I.getOperand(0), I.getOperand(1), Builder))
       return replaceInstUsesWith(I, Res);
 
+    if (Value *Res = foldIntegerRepackThroughZExt(I.getOperand(0),
```

Comment:
Combine this and the `foldBitmaskMul` call above into a single call to `foldDisjointOr`.